keys/keybase.go

```diff
@@ -32,58 +26,59 @@ func New(db dbm.DB, codec words.Codec) dbKeybase {
 
 var _ Keybase = dbKeybase{}
 
-// Create generates a new key and persists it storage, encrypted using the passphrase.
-// It returns the generated seedphrase (mnemonic) and the key Info.
-// It returns an error if it fails to generate a key for the given algo type,
-// or if another key is already stored under the same name.
-func (kb dbKeybase) Create(name, passphrase, algo string) (string, Info, error) {
+// Create generates a new key and persists it to storage, encrypted
+// using the passphrase.  It returns the generated seedphrase
+// (mnemonic) and the key Info.  It returns an error if it fails to
+// generate a key for the given algo type, or if another key is
+// already stored under the same name.
+func (kb dbKeybase) Create(name, passphrase string, algo CryptoAlgo) (Info, string, error) {
 	// NOTE: secret is SHA256 hashed by secp256k1 and ed25519.
 	// 16 byte secret corresponds to 12 BIP39 words.
 	// XXX: Ledgers use 24 words now - should we ?
 	secret := crypto.CRandBytes(16)
-	key, err := generate(algo, secret)
+	priv, err := generate(algo, secret)
 	if err != nil {
-		return "", Info{}, err
+		return Info{}, "", err
 	}
 
 	// encrypt and persist the key
-	public := kb.writeKey(key, name, passphrase)
+	info := kb.writeKey(priv, name, passphrase)
+
+	// we append the type byte to the serialized secret to help with
+	// recovery
+	// ie [secret] = [type] + [secret]
```

Comment:
why do this? it seems wrong to influence the words in any way. Why not just pass in the algo explicitly ? This seems like unecessary unstandard weirdness ...

Reply:
it's a UX thing.  I don't want to have to remember what the words were for.

Reply:
It'll just confused the user.  Maybe we can just prepend a single word that represents the type of key, so it's 13 words?

Reply:
lets do eg. `ed25519{ house table chair ... }`